lib/Auth/Source/OpenIDConnect.php

```diff
@@ -111,9 +117,13 @@ protected function postFinalStep(AccessToken $accessToken, AbstractProvider $pro
     public function logout(&$state)
     {
         $providerLabel = $this->getLabel();
-        $endSessionEndpoint = $this->config->getString('urlEndSession', null);
+        if (array_key_exists('oidc:localLogout', $state) && $state['oidc:localLogout'] === true) {
+            Logger::debug("authoauth2: $providerLabel OP initiated logout");
+            return;
+        }
+        $endSessionEndpoint = $this->getProvider($this->config)->getEndSessionEndpoint();
         if (!$endSessionEndpoint) {
-            Logger::debug("authoauth2: $providerLabel No urlEndSession configured, not doing anything for logout");
+            Logger::debug("authoauth2: $providerLabel OP does not provided and 'end_session_endpoint', not doing anything for logout");
```

Comment:
This should be reworded, I think.
```php
Logger::debug("authoauth2: $providerLabel OP does not provide an 'end_session_endpoint', not doing anything for logout");
```

maybe?

Reply:
indeed.